File: tests/algorithms/x/Clojure/conversions/temperature_conversions.clj
(ns main (:refer-clojure :exclude [floor pow10 round_to celsius_to_fahrenheit celsius_to_kelvin celsius_to_rankine fahrenheit_to_celsius fahrenheit_to_kelvin fahrenheit_to_rankine kelvin_to_celsius kelvin_to_fahrenheit kelvin_to_rankine rankine_to_celsius rankine_to_fahrenheit rankine_to_kelvin reaumur_to_kelvin reaumur_to_fahrenheit reaumur_to_celsius reaumur_to_rankine]))

(require 'clojure.set)

(defn in [x coll]
  (cond (string? coll) (clojure.string/includes? coll x) (map? coll) (contains? coll x) (sequential? coll) (some (fn [e] (= e x)) coll) :else false))

(defn padStart [s w p]
  (loop [out (str s)] (if (< (count out) w) (recur (str p out)) out)))

(defn indexOf [s sub]
  (let [idx (clojure.string/index-of s sub)] (if (nil? idx) -1 idx)))

(defn split [s sep]
  (clojure.string/split s (re-pattern sep)))

(def nowSeed (atom (let [s (System/getenv "MOCHI_NOW_SEED")] (if (and s (not (= s ""))) (Integer/parseInt s) 0))))

(declare floor pow10 round_to celsius_to_fahrenheit celsius_to_kelvin celsius_to_rankine fahrenheit_to_celsius fahrenheit_to_kelvin fahrenheit_to_rankine kelvin_to_celsius kelvin_to_fahrenheit kelvin_to_rankine rankine_to_celsius rankine_to_fahrenheit rankine_to_kelvin reaumur_to_kelvin reaumur_to_fahrenheit reaumur_to_celsius reaumur_to_rankine)

(def ^:dynamic floor_i nil)

(def ^:dynamic pow10_i nil)

(def ^:dynamic pow10_p nil)

(def ^:dynamic round_to_m nil)

(defn floor [floor_x]
  (binding [floor_i nil] (try (do (set! floor_i (long floor_x)) (when (> (double floor_i) floor_x) (set! floor_i (- floor_i 1))) (throw (ex-info "return" {:v (double floor_i)}))) (catch clojure.lang.ExceptionInfo e (if (= (ex-message e) "return") (get (ex-data e) :v) (throw e))))))

(defn pow10 [pow10_n]
  (binding [pow10_i nil pow10_p nil] (try (do (set! pow10_p 1.0) (set! pow10_i 0) (while (< pow10_i pow10_n) (do (set! pow10_p (* pow10_p 10.0)) (set! pow10_i (+ pow10_i 1)))) (throw (ex-info "return" {:v pow10_p}))) (catch clojure.lang.ExceptionInfo e (if (= (ex-message e) "return") (get (ex-data e) :v) (throw e))))))

(defn round_to [round_to_x round_to_ndigits]
  (binding [round_to_m nil] (try (do (set! round_to_m (pow10 round_to_ndigits)) (throw (ex-info "return" {:v (/ (floor (+ (* round_to_x round_to_m) 0.5)) round_to_m)}))) (catch clojure.lang.ExceptionInfo e (if (= (ex-message e) "return") (get (ex-data e) :v) (throw e))))))

(defn celsius_to_fahrenheit [celsius_to_fahrenheit_c celsius_to_fahrenheit_ndigits]
  (try (throw (ex-info "return" {:v (round_to (+ (/ (* celsius_to_fahrenheit_c 9.0) 5.0) 32.0) celsius_to_fahrenheit_ndigits)})) (catch clojure.lang.ExceptionInfo e (if (= (ex-message e) "return") (get (ex-data e) :v) (throw e)))))

(defn celsius_to_kelvin [celsius_to_kelvin_c celsius_to_kelvin_ndigits]
  (try (throw (ex-info "return" {:v (round_to (+ celsius_to_kelvin_c 273.15) celsius_to_kelvin_ndigits)})) (catch clojure.lang.ExceptionInfo e (if (= (ex-message e) "return") (get (ex-data e) :v) (throw e)))))

(defn celsius_to_rankine [celsius_to_rankine_c celsius_to_rankine_ndigits]
  (try (throw (ex-info "return" {:v (round_to (+ (/ (* celsius_to_rankine_c 9.0) 5.0) 491.67) celsius_to_rankine_ndigits)})) (catch clojure.lang.ExceptionInfo e (if (= (ex-message e) "return") (get (ex-data e) :v) (throw e)))))

(defn fahrenheit_to_celsius [fahrenheit_to_celsius_f fahrenheit_to_celsius_ndigits]
  (try (throw (ex-info "return" {:v (round_to (/ (* (- fahrenheit_to_celsius_f 32.0) 5.0) 9.0) fahrenheit_to_celsius_ndigits)})) (catch clojure.lang.ExceptionInfo e (if (= (ex-message e) "return") (get (ex-data e) :v) (throw e)))))

(defn fahrenheit_to_kelvin [fahrenheit_to_kelvin_f fahrenheit_to_kelvin_ndigits]
  (try (throw (ex-info "return" {:v (round_to (+ (/ (* (- fahrenheit_to_kelvin_f 32.0) 5.0) 9.0) 273.15) fahrenheit_to_kelvin_ndigits)})) (catch clojure.lang.ExceptionInfo e (if (= (ex-message e) "return") (get (ex-data e) :v) (throw e)))))

(defn fahrenheit_to_rankine [fahrenheit_to_rankine_f fahrenheit_to_rankine_ndigits]
  (try (throw (ex-info "return" {:v (round_to (+ fahrenheit_to_rankine_f 459.67) fahrenheit_to_rankine_ndigits)})) (catch clojure.lang.ExceptionInfo e (if (= (ex-message e) "return") (get (ex-data e) :v) (throw e)))))

(defn kelvin_to_celsius [kelvin_to_celsius_k kelvin_to_celsius_ndigits]
  (try (throw (ex-info "return" {:v (round_to (- kelvin_to_celsius_k 273.15) kelvin_to_celsius_ndigits)})) (catch clojure.lang.ExceptionInfo e (if (= (ex-message e) "return") (get (ex-data e) :v) (throw e)))))

(defn kelvin_to_fahrenheit [kelvin_to_fahrenheit_k kelvin_to_fahrenheit_ndigits]
  (try (throw (ex-info "return" {:v (round_to (+ (/ (* (- kelvin_to_fahrenheit_k 273.15) 9.0) 5.0) 32.0) kelvin_to_fahrenheit_ndigits)})) (catch clojure.lang.ExceptionInfo e (if (= (ex-message e) "return") (get (ex-data e) :v) (throw e)))))

(defn kelvin_to_rankine [kelvin_to_rankine_k kelvin_to_rankine_ndigits]
  (try (throw (ex-info "return" {:v (round_to (/ (* kelvin_to_rankine_k 9.0) 5.0) kelvin_to_rankine_ndigits)})) (catch clojure.lang.ExceptionInfo e (if (= (ex-message e) "return") (get (ex-data e) :v) (throw e)))))

(defn rankine_to_celsius [rankine_to_celsius_r rankine_to_celsius_ndigits]
  (try (throw (ex-info "return" {:v (round_to (/ (* (- rankine_to_celsius_r 491.67) 5.0) 9.0) rankine_to_celsius_ndigits)})) (catch clojure.lang.ExceptionInfo e (if (= (ex-message e) "return") (get (ex-data e) :v) (throw e)))))

(defn rankine_to_fahrenheit [rankine_to_fahrenheit_r rankine_to_fahrenheit_ndigits]
  (try (throw (ex-info "return" {:v (round_to (- rankine_to_fahrenheit_r 459.67) rankine_to_fahrenheit_ndigits)})) (catch clojure.lang.ExceptionInfo e (if (= (ex-message e) "return") (get (ex-data e) :v) (throw e)))))

(defn rankine_to_kelvin [rankine_to_kelvin_r rankine_to_kelvin_ndigits]
  (try (throw (ex-info "return" {:v (round_to (/ (* rankine_to_kelvin_r 5.0) 9.0) rankine_to_kelvin_ndigits)})) (catch clojure.lang.ExceptionInfo e (if (= (ex-message e) "return") (get (ex-data e) :v) (throw e)))))

(defn reaumur_to_kelvin [reaumur_to_kelvin_r reaumur_to_kelvin_ndigits]
  (try (throw (ex-info "return" {:v (round_to (+ (* reaumur_to_kelvin_r 1.25) 273.15) reaumur_to_kelvin_ndigits)})) (catch clojure.lang.ExceptionInfo e (if (= (ex-message e) "return") (get (ex-data e) :v) (throw e)))))

(defn reaumur_to_fahrenheit [reaumur_to_fahrenheit_r reaumur_to_fahrenheit_ndigits]
  (try (throw (ex-info "return" {:v (round_to (+ (* reaumur_to_fahrenheit_r 2.25) 32.0) reaumur_to_fahrenheit_ndigits)})) (catch clojure.lang.ExceptionInfo e (if (= (ex-message e) "return") (get (ex-data e) :v) (throw e)))))

(defn reaumur_to_celsius [reaumur_to_celsius_r reaumur_to_celsius_ndigits]
  (try (throw (ex-info "return" {:v (round_to (* reaumur_to_celsius_r 1.25) reaumur_to_celsius_ndigits)})) (catch clojure.lang.ExceptionInfo e (if (= (ex-message e) "return") (get (ex-data e) :v) (throw e)))))

(defn reaumur_to_rankine [reaumur_to_rankine_r reaumur_to_rankine_ndigits]
  (try (throw (ex-info "return" {:v (round_to (+ (+ (* reaumur_to_rankine_r 2.25) 32.0) 459.67) reaumur_to_rankine_ndigits)})) (catch clojure.lang.ExceptionInfo e (if (= (ex-message e) "return") (get (ex-data e) :v) (throw e)))))

(defn -main []
  (let [rt (Runtime/getRuntime)
    start-mem (- (.totalMemory rt) (.freeMemory rt))
    start (System/nanoTime)]
      (println (celsius_to_fahrenheit 0.0 2))
      (println (celsius_to_kelvin 0.0 2))
      (println (celsius_to_rankine 0.0 2))
      (println (fahrenheit_to_celsius 32.0 2))
      (println (fahrenheit_to_kelvin 32.0 2))
      (println (fahrenheit_to_rankine 32.0 2))
      (println (kelvin_to_celsius 273.15 2))
      (println (kelvin_to_fahrenheit 273.15 2))
      (println (kelvin_to_rankine 273.15 2))
      (println (rankine_to_celsius 491.67 2))
      (println (rankine_to_fahrenheit 491.67 2))
      (println (rankine_to_kelvin 491.67 2))
      (println (reaumur_to_kelvin 80.0 2))
      (println (reaumur_to_fahrenheit 80.0 2))
      (println (reaumur_to_celsius 80.0 2))
      (println (reaumur_to_rankine 80.0 2))
      (System/gc)
      (let [end (System/nanoTime)
        end-mem (- (.totalMemory rt) (.freeMemory rt))
        duration-us (quot (- end start) 1000)
        memory-bytes (Math/abs ^long (- end-mem start-mem))]
        (println (str "{\n  \"duration_us\": " duration-us ",\n  \"memory_bytes\": " memory-bytes ",\n  \"name\": \"main\"\n}"))
      )
    ))

(-main)

File: tests/algorithms/transpiler/FS/conversions/temperature_conversions.fs
// Generated 2025-08-07 10:31 +0700

exception Return
let mutable _nowSeed:int64 = 0L
let mutable _nowSeeded = false
let _initNow () =
    let s = System.Environment.GetEnvironmentVariable("MOCHI_NOW_SEED")
    if System.String.IsNullOrEmpty(s) |> not then
        match System.Int32.TryParse(s) with
        | true, v ->
            _nowSeed <- int64 v
            _nowSeeded <- true
        | _ -> ()
let _now () =
    if _nowSeeded then
        _nowSeed <- (_nowSeed * 1664525L + 1013904223L) % 2147483647L
        int _nowSeed
    else
        int (System.DateTime.UtcNow.Ticks % 2147483647L)

_initNow()
let __bench_start = _now()
let __mem_start = System.GC.GetTotalMemory(true)
let rec floor (x: float) =
    let mutable __ret : float = Unchecked.defaultof<float>
    let mutable x = x
    try
        let mutable i: int = int x
        if (float i) > x then
            i <- i - 1
        __ret <- float i
        raise Return
        __ret
    with
        | Return -> __ret
let rec pow10 (n: int) =
    let mutable __ret : float = Unchecked.defaultof<float>
    let mutable n = n
    try
        let mutable p: float = 1.0
        let mutable i: int = 0
        while i < n do
            p <- p * 10.0
            i <- i + 1
        __ret <- p
        raise Return
        __ret
    with
        | Return -> __ret
let rec round_to (x: float) (ndigits: int) =
    let mutable __ret : float = Unchecked.defaultof<float>
    let mutable x = x
    let mutable ndigits = ndigits
    try
        let m: float = pow10 (ndigits)
        __ret <- (floor ((x * m) + 0.5)) / m
        raise Return
        __ret
    with
        | Return -> __ret
let rec celsius_to_fahrenheit (c: float) (ndigits: int) =
    let mutable __ret : float = Unchecked.defaultof<float>
    let mutable c = c
    let mutable ndigits = ndigits
    try
        __ret <- round_to (((c * 9.0) / 5.0) + 32.0) (ndigits)
        raise Return
        __ret
    with
        | Return -> __ret
let rec celsius_to_kelvin (c: float) (ndigits: int) =
    let mutable __ret : float = Unchecked.defaultof<float>
    let mutable c = c
    let mutable ndigits = ndigits
    try
        __ret <- round_to (c + 273.15) (ndigits)
        raise Return
        __ret
    with
        | Return -> __ret
let rec celsius_to_rankine (c: float) (ndigits: int) =
    let mutable __ret : float = Unchecked.defaultof<float>
    let mutable c = c
    let mutable ndigits = ndigits
    try
        __ret <- round_to (((c * 9.0) / 5.0) + 491.67) (ndigits)
        raise Return
        __ret
    with
        | Return -> __ret
let rec fahrenheit_to_celsius (f: float) (ndigits: int) =
    let mutable __ret : float = Unchecked.defaultof<float>
    let mutable f = f
    let mutable ndigits = ndigits
    try
        __ret <- round_to (((f - 32.0) * 5.0) / 9.0) (ndigits)
        raise Return
        __ret
    with
        | Return -> __ret
let rec fahrenheit_to_kelvin (f: float) (ndigits: int) =
    let mutable __ret : float = Unchecked.defaultof<float>
    let mutable f = f
    let mutable ndigits = ndigits
    try
        __ret <- round_to ((((f - 32.0) * 5.0) / 9.0) + 273.15) (ndigits)
        raise Return
        __ret
    with
        | Return -> __ret
let rec fahrenheit_to_rankine (f: float) (ndigits: int) =
    let mutable __ret : float = Unchecked.defaultof<float>
    let mutable f = f
    let mutable ndigits = ndigits
    try
        __ret <- round_to (f + 459.67) (ndigits)
        raise Return
        __ret
    with
        | Return -> __ret
let rec kelvin_to_celsius (k: float) (ndigits: int) =
    let mutable __ret : float = Unchecked.defaultof<float>
    let mutable k = k
    let mutable ndigits = ndigits
    try
        __ret <- round_to (k - 273.15) (ndigits)
        raise Return
        __ret
    with
        | Return -> __ret
let rec kelvin_to_fahrenheit (k: float) (ndigits: int) =
    let mutable __ret : float = Unchecked.defaultof<float>
    let mutable k = k
    let mutable ndigits = ndigits
    try
        __ret <- round_to ((((k - 273.15) * 9.0) / 5.0) + 32.0) (ndigits)
        raise Return
        __ret
    with
        | Return -> __ret
let rec kelvin_to_rankine (k: float) (ndigits: int) =
    let mutable __ret : float = Unchecked.defaultof<float>
    let mutable k = k
    let mutable ndigits = ndigits
    try
        __ret <- round_to ((k * 9.0) / 5.0) (ndigits)
        raise Return
        __ret
    with
        | Return -> __ret
let rec rankine_to_celsius (r: float) (ndigits: int) =
    let mutable __ret : float = Unchecked.defaultof<float>
    let mutable r = r
    let mutable ndigits = ndigits
    try
        __ret <- round_to (((r - 491.67) * 5.0) / 9.0) (ndigits)
        raise Return
        __ret
    with
        | Return -> __ret
let rec rankine_to_fahrenheit (r: float) (ndigits: int) =
    let mutable __ret : float = Unchecked.defaultof<float>
    let mutable r = r
    let mutable ndigits = ndigits
    try
        __ret <- round_to (r - 459.67) (ndigits)
        raise Return
        __ret
    with
        | Return -> __ret
let rec rankine_to_kelvin (r: float) (ndigits: int) =
    let mutable __ret : float = Unchecked.defaultof<float>
    let mutable r = r
    let mutable ndigits = ndigits
    try
        __ret <- round_to ((r * 5.0) / 9.0) (ndigits)
        raise Return
        __ret
    with
        | Return -> __ret
let rec reaumur_to_kelvin (r: float) (ndigits: int) =
    let mutable __ret : float = Unchecked.defaultof<float>
    let mutable r = r
    let mutable ndigits = ndigits
    try
        __ret <- round_to ((r * 1.25) + 273.15) (ndigits)
        raise Return
        __ret
    with
        | Return -> __ret
let rec reaumur_to_fahrenheit (r: float) (ndigits: int) =
    let mutable __ret : float = Unchecked.defaultof<float>
    let mutable r = r
    let mutable ndigits = ndigits
    try
        __ret <- round_to ((r * 2.25) + 32.0) (ndigits)
        raise Return
        __ret
    with
        | Return -> __ret
let rec reaumur_to_celsius (r: float) (ndigits: int) =
    let mutable __ret : float = Unchecked.defaultof<float>
    let mutable r = r
    let mutable ndigits = ndigits
    try
        __ret <- round_to (r * 1.25) (ndigits)
        raise Return
        __ret
    with
        | Return -> __ret
let rec reaumur_to_rankine (r: float) (ndigits: int) =
    let mutable __ret : float = Unchecked.defaultof<float>
    let mutable r = r
    let mutable ndigits = ndigits
    try
        __ret <- round_to (((r * 2.25) + 32.0) + 459.67) (ndigits)
        raise Return
        __ret
    with
        | Return -> __ret
printfn "%g" (celsius_to_fahrenheit (0.0) (2))
printfn "%g" (celsius_to_kelvin (0.0) (2))
printfn "%g" (celsius_to_rankine (0.0) (2))
printfn "%g" (fahrenheit_to_celsius (32.0) (2))
printfn "%g" (fahrenheit_to_kelvin (32.0) (2))
printfn "%g" (fahrenheit_to_rankine (32.0) (2))
printfn "%g" (kelvin_to_celsius (273.15) (2))
printfn "%g" (kelvin_to_fahrenheit (273.15) (2))
printfn "%g" (kelvin_to_rankine (273.15) (2))
printfn "%g" (rankine_to_celsius (491.67) (2))
printfn "%g" (rankine_to_fahrenheit (491.67) (2))
printfn "%g" (rankine_to_kelvin (491.67) (2))
printfn "%g" (reaumur_to_kelvin (80.0) (2))
printfn "%g" (reaumur_to_fahrenheit (80.0) (2))
printfn "%g" (reaumur_to_celsius (80.0) (2))
printfn "%g" (reaumur_to_rankine (80.0) (2))
let __bench_end = _now()
let __mem_end = System.GC.GetTotalMemory(true)
printfn "{\n  \"duration_us\": %d,\n  \"memory_bytes\": %d,\n  \"name\": \"main\"\n}" ((__bench_end - __bench_start) / 1000) (__mem_end - __mem_start)
